app/models/league.rb

```diff
@@ -18,53 +17,58 @@ def to_param
   end
 
   def update_badges
-    self.users.update_all(most_wins: false, top_crawler: false, worst_crawler: false, longest_winning_streak: false, most_teams: false, longest_winning_streak_ever: false )
-
-    most_wins.update_attribute(:most_wins, true) if most_wins && most_wins.number_of_wins > 0
-    top_crawler.update_attribute(:top_crawler, true) if top_crawler && top_crawler.number_of_crawls > 0
-    worst_crawler.update_attribute(:worst_crawler, true) if worst_crawler && worst_crawler.number_of_crawlings > 0
-    longest_winning_streak.update_attribute(:longest_winning_streak, true) if longest_winning_streak && longest_winning_streak.winning_streak > 0
-    most_teams.update_attribute(:most_teams, true) if most_teams
-    longest_winning_streak_ever.update_attribute(:longest_winning_streak_ever, true) if longest_winning_streak_ever
-
+    reset_all_user_badges
+    ensure_user(most_wins).update_attribute(:most_wins, true)
+    ensure_user(top_crawler).update_attribute(:top_crawler, true)
+    ensure_user(worst_crawler).update_attribute(:worst_crawler, true)
+    ensure_user(longest_winning_streak).update_attribute(:longest_winning_streak, true)
+    ensure_user(most_teams).update_attribute(:most_teams, true)
+    ensure_user(longest_winning_streak_ever).update_attribute(:longest_winning_streak_ever, true)
```

Comment:
is it on intend that we skip all validations with `#update_attribute`? If not I would prefer just `#update`